daemon/mgr/container_types.go

```diff
@@ -352,6 +355,46 @@ func (c *Container) UnsetMergedDir() {
 	c.Snapshotter.Data["MergedDir"] = ""
 }
 
+// SetSnapshotterMeta sets snapshotter for container
+func (c *Container) SetSnapshotterMeta(mounts []mount.Mount) {
+	// TODO(ziren): now we only support overlayfs
+	data := make(map[string]string, 0)
+	for _, opt := range mounts[0].Options {
+		if strings.HasPrefix(opt, "upperdir=") {
+			data["UpperDir"] = strings.TrimPrefix(opt, "upperdir=")
+		}
+		if strings.HasPrefix(opt, "lowerdir=") {
+			data["LowerDir"] = strings.TrimPrefix(opt, "lowerdir=")
+		}
+		if strings.HasPrefix(opt, "workdir=") {
+			data["WorkDir"] = strings.TrimPrefix(opt, "workdir=")
+		}
+	}
+
+	c.Snapshotter = &types.SnapshotterData{
+		Name: "overlayfs",
+		Data: data,
+	}
+}
+
+// GetSpecificBasePath accepts a given path, look for whether the path is exist
+// within container, if has, returns container base path like BaseFS, if not, return empty string
+func (c *Container) GetSpecificBasePath(path string) string {
```

Comment:
feel weird ! can we judge whether the baseFS is mounted?

Reply:
i still have another question: can we ensure passwd file exists in UpperDir? 

Or can we mount the mergedDir when we found it not mounted, then we can ensure the passwd file exists in MergedDir.

it's just my advice, you can think about it again, thanks a lot @Ace-Tang

Reply:
when we run a container, merged dir is not created yet until pouch pass create options to containerd, then containerd created it and mount. Add this function cause we can only find file in image.

Reply:
I also think this pr can fix if merged dir is un-mounted by error.